source/core/Events.cpp

```diff
@@ -160,7 +160,7 @@ class EventQueueObject {
 
 typedef std::vector<EventQueueObject> EventQueueObjectVector;
 
-typedef Int32 EventOracleIndex;
+typedef IntIndex EventOracleIndex;
```

Comment:
Use size_t, instead of IntIndex and remove the static_casts, do the same for EventQueueID and remove the casts introduced for qID.

Reply:
Disagree, size_t is no good because:
  1) it's unsigned, but the sentinel kNotAnEventOracleIndex is -1.  Zero is a valid EventOracleIndex.
   We could change this (not much code relies on it), but this usage matches NCG.
 2) It's 8 bytes on native 64-bit platforms; I want it to be the same size everywhere.

Reply:
Ok, I wasn't aware of the negative value.

Reply:
I'm deferring review of changes to this file (Events.cpp) to @spathiwa